pkg/cmd/util/downloadrequest/downloadrequest.go

```diff
@@ -99,11 +99,27 @@ Loop:
 		return ErrNotFound
 	}
 
-	httpClient := new(http.Client)
-	if insecureSkipTLSVerify {
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	var caPool *x509.CertPool
+	if len(caCertPath) > 0 {
+		caCert, err := ioutil.ReadFile(caCertPath)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't open cacert")
+		}
+		// bundle the passed in cert with the system cert pool
+		// if it's available, otherwise create a new pool just
+		// for this.
+		caPool, err = x509.SystemCertPool()
+		if err != nil {
+			caPool = x509.NewCertPool()
 		}
+		caPool.AppendCertsFromPEM(caCert)
+	}
+	httpClient := new(http.Client)
+	httpClient.Transport = &http.Transport{
```

Comment:
I think we probably want to use the timeout passed to `Stream` on the transport, since the Go HTTP client doesn't have one by default and will just wait forever if it's not specified (https://golang.org/src/net/http/transport.go#L211).

Reply:
Seems like the right call. While fixing that I also set the rest of the values on the transport to match go's DefaultTransport. https://golang.org/src/net/http/transport.go#L42